xscen/indicators.py

```diff
@@ -127,7 +127,19 @@ def compute_indicators(
             out = ind(ds=ds)
 
             # Infer the indicator's frequency
-            freq = xr.infer_freq(out.time) if "time" in out.dims else "fx"
+            if "time" in out.dims:
+                if len(out.time) < 3:
+                    freq = (
+                        ind.injected_parameters["freq"]
+                        if "freq" in ind.injected_parameters
+                        else ind.parameters["freq"]["default"]
+                        if "freq" in ind.parameters
+                        else ind.src_freq
+                    )
+                else:
+                    freq = xr.infer_freq(out.time)
+            else:
+                freq = "fx"
```

Comment:
Do you think ii would be better to wrap this in a private function instead of having it twice?